Makefile

```diff
@@ -0,0 +1,180 @@
+# *
+# * Copyright 2023 RDK Management
+# *
+# * Licensed under the Apache License, Version 2.0 (the "License");
+# * you may not use this file except in compliance with the License.
+# * You may obtain a copy of the License at
+# *
+# * http://www.apache.org/licenses/LICENSE-2.0
+# *
+# * Unless required by applicable law or agreed to in writing, software
+# * distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+# *
+
+TARGET_DYNAMIC_LIB ?= libutcontrollibrary.so
+TARGET_STATIC_LIB ?= libutcontrollibrary.a
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[0;33m'
+NC='\033[0m'
+
+$(info $(shell echo -e ${GREEN}TARGET_DYNAMIC_LIB [$(TARGET_DYNAMIC_LIB)]${NC}))
+$(info $(shell echo -e ${GREEN}TARGET_STATIC_LIB [$(TARGET_STATIC_LIB)]${NC}))
+$(info $(shell echo -e ${GREEN}TARGET_BIN [$(TARGET_BIN)]${NC}))
+
+UT_CONTROL_DIR:=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+export PATH := $(shell pwd)/toolchain:$(PATH)
+TOP_DIR ?= $(UT_CONTROL_DIR)
+BUILD_DIR ?= $(TOP_DIR)/obj
+BIN_DIR ?= $(TOP_DIR)/bin
+LIB_DIR := $(TOP_DIR)/lib
+LIB_TEST_DIR := $(TOP_DIR)/tests/lib
+BUILD_LIBS = yes
+
+ifneq ($(TARGET_BIN),ut-control-test)
```

Comment:
?? Why would the requirements for the test be different than the standard requirements?

The is not the test makefile, this is the main makefile... the test requires it's own.

Reply:
Yes, thats true. However, when the test builds it only needs it core src files and we are linking the ut_control.so against the test, rather than building all of them as a library.
If we are ok to build it completely as a binary then I shall remove this check.
My thought process was to prove that this library works fine against test so it should be ok for everyone else.
But, we can have it other ways also.